### ISSUE TITLE
fyi: Referential Integrity Problem

### ISSUE THREAD

**Issue by vonglan** [SAP/abap-cleaner#322]
For your information - not sure if this can be solved:

For this code:

```
REPORT zevg.

CLASS lcl_app DEFINITION FINAL.
  PUBLIC SECTION.
    CLASS-METHODS:
      main.
  PRIVATE SECTION.
ENDCLASS.

CLASS lcl_app IMPLEMENTATION.
  METHOD main.
    DO 4 TIMES.
      CASE sy-index.
        WHEN 1 OR 2 OR 3. DATA(ext) = |AV{ sy-index - 1 }|.
        WHEN OTHERS. ext = 'MAX'.
      ENDCASE.
"      WRITE |now using { ext }|.
    ENDDO.
  ENDMETHOD.
ENDCLASS.

START-OF-SELECTION.
  lcl_app=>main( ).
```
I get the message:

`Error` executing rule 'Delete unused variables': Failed referential integrity test on command starting at source line 14!

When I add a usage of the variable, the problem disappears.

**Comment by jmgrassau** [comment]
Hi Edo, 

thanks for reporting this! This seems to be similar to #315: The issue is gone if you split this line into two:

```ABAP
        WHEN 1 OR 2 OR 3. 
          DATA(ext) = |AV{ sy-index - 1 }|.
```

From the first look of it, this happens when ABAP cleaner tries to put a TODO comment ("variable is assigned but never used") before the DATA(ext) statement, and doesn't properly consider that the statement follows the previous statement on the same line.

Kind regards,
Jörg-Michael

**Comment by vonglan** [comment]
Hi Jörg-Michael,
yes, looks similar.
Not a real problem.
Thanks,
Edo

**Comment by jmgrassau** [comment]
Hi Edo, 

this bug will be fixed with the next release. Since the comment cannot be inserted on the same line between the two statements, the statements are moved to an own line: 

![image](https://github.com/SAP/abap-cleaner/assets/79639298/4368a8b2-ef77-4122-8855-89761228e757)

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
Hi Edo,

this bug should now be fixed with version 1.17.1, which was just released! Thanks again for reporting this.

Kind regards,
Jörg-Michael
